### PR TITLE
TOOLS-2315 document how to add to /Joyent_Dev/public/releng/go/adhoc for new go versions

### PR DESCRIPTION
**Pull request by timfoster**

### CODE REVIEW

File: tools/releng/go-prebuilt/go-build.sh [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+# $ ~/projects/triton.git/tools/releng/go-prebuilt/go-build.sh \
+#      -N 1.14 -b 1.13rc1 -d /ws/gobuild
+#
+#
+
+if [[ -n "$TRACE" ]]; then
+    export PS4='${BASH_SOURCE}:${LINENO}: '
+    set -o xtrace
+fi
+set -o errexit
```
Comment (trentm) [inline tools/releng/go-prebuilt/go-build.sh]: Also `set -o pipefail` perhaps?

File: tools/releng/go-prebuilt/go-build.sh [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    fi
+
+    # Remove the bootstrap compiler files.
+    rm -rf "pkg/bootstrap"
+
+    # Remove the obj build cache, which is _huge_.
+    rm -rf "pkg/obj"
+
+    # Remove the libraries used to build the toolchain commands.
+    rm -rf "pkg/illumos_amd64/cmd"
```
Comment (trentm) [inline tools/releng/go-prebuilt/go-build.sh]: This go-build.sh only needs to support newer Go versions that use the "illumos" GOOS, right?

File: tools/releng/go-prebuilt/go-build.sh (hunk 1 above)
Comment (timfoster) [reply tools/releng/go-prebuilt/go-build.sh]: Good idea, thanks

File: tools/releng/go-prebuilt/go-build.sh (hunk 2 above)
Comment (timfoster) [reply tools/releng/go-prebuilt/go-build.sh]: Yes - I'm assuming here that we always only want to build a go runtime for illumos. That's certainly the platform we'll be running on, and given that a go runtime will cross-compile, if we want to declare 'solaris' when building any Triton/Manta component at build-time, we can do eg.
```
GOOS=solaris go build
```
even with our illumos binary. For now, that's actually the default in Makefile.go_prebuilt.targ. I mentioned that in TOOLS-2439.

**Pull request merged**